enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,503 @@
+permissions, and with this CCO could be the alerting mechanism for a changed CredentialsRequest without sufficient 
+permissions.
+
+**Pod Identity Webhook mode** (Deployed on classic ROSA, ROSA on HyperShift, standard OCP and Hypershift)
+
+Will work by annotating the ServiceAccount triggering the projection of the service account tokens into pods created for
+the operator (same method and resultant actions as: https://github.com/openshift/hypershift/pull/1351) the README here 
+gives an AWS specific example: https://github.com/openshift/aws-pod-identity-webhook)
+
+Possibly some changes to logic for Pod Identity Webhooks as needed for fine-grained credential
+management to allow for multi-tenancy (see definition earlier). Also, need to add pod identity webhooks targeting other
+cloud providers and change the annotations to more generic naming.
+
+**HyperShift changes**: Include Cloud Credential Operator with "Token" mode (see above). Allows for processing of 
+CredentialsRequest objects added by operators.
+
+**OperatorHub and Console changes**: Allow for import of additional, well-known, bundle annotations which will result 
+in ENV variables on the Subscription object. Show operator is enabled for Token-based use based on Console determining 
+cluster is running in token-based authentication enabled cluster by reading the `.spec.serviceAccountIssuer` from the Authentication CR, `.spec.platformSpec.type` from the Infrastructure CR,`.spec.credentialsMode` from the CloudCredentials CR  AND
+bundle has token-based auth enabled annotation. This info should be written to a new field on Subscription objects.
+The Subscription fields will allow UX for the information needed by CCO or webhook, for input of the cloud credentials. 
+Generate a manual ack for operator upgrade when cloud resources are involved. This can be determined by parsing the 
+Subscription objects. Ensures admin signs off that cloud resources are provisioned and working as intended before an 
+operator upgrade.
+
+**Operator team/ Operator SDK changes**: Follow new guidelines for allowing for the operator to work on token auth 
+enabled cluster. New guidelines would include the following to use CCO Token mode:
+
+- changing CRD to add CredentialRequest references and putting those referenced
+- CredentialRequests into a defined directory in the bundle
+- Add a bundle annotation claiming token-based authentication support
+- add the projected ServiceAccount volume to deployment;
+- handle the Secret themselves (like today’s CCO Mint mode, read and use the credentials in the Secret whose name they 
+  know from the CredentialsRequest and is in the same Namespace)
+- Guidance to operator authors to structure controller code to alert on missing cloud credentials, not crash.
+
+SDK to support this new template. SDK to validate and in particular: alert on any permission changes between operator
+versions.
+
+### Workflow Description
```

Comment:
lgtm

Reply:
two issues w/ the above flow:

1) the operator pod can't mount the secret because it doesn't exist until the operator has come up.  Joe and I discussed this and there are various flows that could be done w/ init containers, or you can just punt and force the operator to wait for + read the secret directly.  None are great answers.  But you can really only avoid that problem if something else (console, OLM) creates the credreq/secret in order to ensure it exists when the deployment is created.

2) my understanding is that step(3) needs to be made transparent in the future (meaning somehow the roles the operator needs are communicated to one of the tools/components involved in the install, so it can create the necessarily roles).  But that may not be an obstacle to phase 1.

Reply:
added